src/compiler/bundle/ext-transforms-plugin.ts

```diff
@@ -16,12 +27,45 @@ export const extTransformsPlugin = (
   return {
     name: 'extTransformsPlugin',
 
+    /**
+     * A custom function targeting the `transform` build hook in Rollup. See here for details:
+     * https://rollupjs.org/guide/en/#transform
+     *
+     * This Rollup build hook has a signature like:
+     *
+     * ```
+     * (code: string, id: string) =>
+     *   | string
+     *   | null
+     *   | {
+     *       code?: string,
+     *       map?: string | SourceMap,
+     *       ast? : ESTree.Program,
+     *       moduleSideEffects?: boolean | "no-treeshake" | null,
+     *       syntheticNamedExports?: boolean | string | null,
+     *       meta?: {[plugin: string]: any} | null
+     *     }
+     * ```
+     *
+     * Here we are ignoring the `code` argument and only looking at the `id` argument.
+     * We use that `id` to get information about the module in question from disk
+     * ourselves so that we can then do some transformations on it.
+     *
+     * @param _ an unused parameter (normally the code for a given module)
+     * @param id the id of a module
+     * @returns metadata for Rollup or null if no transformation should be done
+     */
     async transform(_, id) {
       if (/\0/.test(id)) {
         return null;
       }
 
+      // The `id` here was possibly previously updated using
+      // `serializeImportPath` to annotate the filepath with various metadata
+      // serialized to query-params. If that was done for this particular `id`
+      // then the `data` prop will not be null.
```

Comment:
I was thinking it might be a helpful note to leave in - when I found `serializeImportPath` and `parseImportPath` and read them I thought it was an odd and sort of surprising thing so thought it might be helpful to leave a little signpost behind

Reply:
I think the sort of strange bit (relating to the thread below about which files are actually written) is that the `id` for a module which has a Stencil component in it will be rewritten using that `serializeImportPath` function to include the information set in the `@Component` decorator as query-params like `?scoped=true&tag=my-component` and so on. This seems like a very indirect and confusing pattern to me since then this information is crucial here for figuring out when we need to write the transformed CSS corresponding to that module...anywhoo just a weird little corner here where I don't quite understand why it's like this